gobblin-core/src/main/java/gobblin/publisher/TimePartitionedDataWithRowCountsPublisher.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (C) 2014-2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.publisher;
+
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.configuration.State;
+import gobblin.configuration.WorkUnitState;
+import gobblin.util.ForkOperatorUtils;
+import gobblin.util.WriterUtils;
+import gobblin.writer.WriterOutputFormat;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.Path;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * For time partition jobs, writer output directory is
+ * $GOBBLIN_WORK_DIR/task-output/{extractId}/{tableName}/{partitionPath},
+ * where partition path is the time bucket, e.g., 2015/04/08/15.
+ *
+ * While moving the file, also places the row count (number of entries contained within the file)
+ * into the name of the file such that output.avro becomes output.{rowCount}.avro.
+ *
+ * Unlike {@link TimePartitionedDataPublisher}, moves each file individually rather than moving
+ * directories at a time. This is done so that each task/branch can move the file it created,
+ * allowing it to access the row count information and place it into the file name.
+ *
+ * Publisher output directory is $GOBBLIN_WORK_DIR/job-output/{tableName}/{partitionPath}
+ *
+ * @author ekrogen
+ */
+public class TimePartitionedDataWithRowCountsPublisher extends TimePartitionedDataPublisher {
```

Comment:
Does this class only work for jobs that write to time partitioned folders?
I guess it can extend `BaseDataPublisher` and be applied to all jobs, correct?